contracts/whitelist/src/contract.rs

```diff
@@ -105,24 +108,21 @@ pub fn instantiate(
     CONFIG.save(deps.storage, &config)?;
 
     if msg.start_time > msg.end_time {
-        return Err(ContractError::InvalidStartTime(
-            msg.start_time,
-            msg.end_time,
-        ));
+        return Err(ContractError::InvalidStartTime(start_time, end_time));
     }
 
-    if msg.start_time.is_expired(&env.block) {
+    if start_time.is_expired(&env.block) {
         return Err(ContractError::InvalidStartTime(
             Expiration::AtTime(env.block.time),
-            msg.start_time,
+            start_time,
         ));
     }
 
-    let genesis_start_time = Expiration::AtTime(Timestamp::from_nanos(GENESIS_MINT_START_TIME));
+    let genesis_start_time = Timestamp::from_nanos(GENESIS_MINT_START_TIME);
     if msg.start_time < genesis_start_time {
         return Err(ContractError::InvalidStartTime(
-            msg.start_time,
-            genesis_start_time,
+            start_time,
+            Expiration::AtTime(genesis_start_time),
         ));
     }
```

Comment:
a bit confused of the mix in using expiration and timestamps, maybe adding comments on which one is being used for new readers of the code.